plugins/inputs/socketstat/socketstat.go

```diff
@@ -0,0 +1,219 @@
+	// a new measurement or postpone it to a later line.
+
+	// The first line is only headers
+	scanner.Scan()
+
+	flushData := false
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line == "" {
+			continue
+		}
+		words := strings.Fields(line)
+
+		var err error
+		if !ss.BeginsWithBlank.MatchString(line) {
+			// A line with no starting whitespace means we're going to parse a new connection.
+			// Flush what we gathered about the previous one, if any.
+			if flushData {
+				acc.AddFields(ss.Measurement, fields, tags)
+			}
+
+			// Delegate the real parsing to getTagsAndState, which manages various
+			// formats depending on the protocol.
+			tags, fields = getTagsAndState(proto, words, ss.Log)
+
+			// This line containted metrics, so record that.
+			flushData = true
+		} else {
+			// A line with starting whitespace means metrics about the current connection.
+			// We should never get 2 consecutive such lines. If we do, log a warning and in
+			// a best effort, extend the metrics from the 1st line with the metrics of the 2nd
+			// one, possibly overwriting.
+			for _, word := range words {
+				if ss.ValidValues.MatchString(word) {
+					// kv will have 2 fields because it matched the regexp
+					kv := strings.Split(word, ":")
+					fields[kv[0]], err = strconv.ParseUint(kv[1], 10, 64)
+					if err != nil {
+						ss.Log.Infof("Couldn't parse metric: %s", word)
+						continue
+					}
+				}
+			}
+			if !flushData {
+				ss.Log.Warnf("Found orphaned metrics: %s", words)
+				ss.Log.Warnf("Added them to the last known connection.")
+			}
+			acc.AddFields(ss.Measurement, fields, tags)
+			flushData = false
+		}
+	}
+	if flushData {
+		acc.AddFields(ss.Measurement, fields, tags)
+	}
+	return nil
+}
+
+func getTagsAndState(proto string, words []string, log telegraf.Logger) (map[string]string, map[string]interface{}) {
+	tags := map[string]string{}
+	fields := make(map[string]interface{})
+	tags["proto"] = proto
+	switch proto {
+	case "udp", "raw":
+		words = append([]string{"dummy"}, words...)
+	case "tcp", "dccp", "sctp":
+		fields["state"] = words[0]
+	}
+	switch proto {
+	case "tcp", "udp", "raw", "dccp", "sctp":
+		// Local and remote addresses are fields 3 and 4
+		// Separate addresses and ports with the last ':'
+		localIndex := strings.LastIndex(words[3], ":")
+		remoteIndex := strings.LastIndex(words[4], ":")
+		tags["local_addr"] = words[3][:localIndex]
+		tags["local_port"] = words[3][localIndex+1:]
+		tags["remote_addr"] = words[4][:remoteIndex]
+		tags["remote_port"] = words[4][remoteIndex+1:]
+	case "unix", "packet":
+		fields["netid"] = words[0]
+		tags["local_addr"] = words[4]
+		tags["local_port"] = words[5]
+		tags["remote_addr"] = words[6]
+		tags["remote_port"] = words[7]
+	}
+	var err error
+	fields["recv_q"], err = strconv.ParseUint(words[1], 10, 64)
+        if err != nil {
+                log.Infof("Couldn't read recv_q in: %s", words)
```

Comment:
Is this really an "info"? I think it's at least a warning or an error as the resulting metric has a missing field.

Reply:
Fixed.